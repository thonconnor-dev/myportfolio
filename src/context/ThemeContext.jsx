import React, { createContext, useContext, useState } from 'react';

const ThemeContext = createContext();

export function useTheme() {
    return useContext(ThemeContext);
}

export function ThemeProvider({ children }) {
    const [isDark, setIsDark] = useState(true);

    const toggleTheme = () => {
        setIsDark(!isDark);
    };

    return (
        <ThemeContext.Provider value={{ isDark, toggleTheme }}>
            {children}
        </ThemeContext.Provider>
    );
}