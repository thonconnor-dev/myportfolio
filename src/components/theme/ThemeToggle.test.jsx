import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import ThemeToggle from './ThemeToggle';
import * as ThemeContextModule from '../../context/ThemeContext';

// Mock the useTheme hook
vi.mock('../../context/ThemeContext', () => ({
  useTheme: vi.fn(),
  ThemeProvider: ({ children }) => <>{children}</>
}));

describe('ThemeToggle', () => {
  it('renders correctly in dark mode', () => {
    // Setup mock return value for useTheme
    const toggleTheme = vi.fn();
    vi.mocked(ThemeContextModule.useTheme).mockReturnValue({ 
      isDark: true, 
      toggleTheme 
    });
    
    const { container } = render(<ThemeToggle />);
    
    // Create a snapshot for dark mode
    expect(container).toMatchSnapshot();
    
    // Additional assertions for specific elements
    const button = screen.getByRole('button', { name: /switch to light mode/i });
    expect(button).toBeInTheDocument();
    expect(button.className).toContain('bg-gray-800');
    expect(button.className).toContain('text-white');
    
    const svg = button.querySelector('svg');
    expect(svg).toBeInTheDocument();
    const path = svg.querySelector('path');
    expect(path.getAttribute('d')).toContain('M12 3v1m0 16v1m9-9h-1M4 12H3m15.364');
  });
  
  it('renders correctly in light mode', () => {
    // Setup mock return value for useTheme
    const toggleTheme = vi.fn();
    vi.mocked(ThemeContextModule.useTheme).mockReturnValue({ 
      isDark: false, 
      toggleTheme 
    });
    
    const { container } = render(<ThemeToggle />);
    
    // Create a snapshot for light mode
    expect(container).toMatchSnapshot();
    
    // Additional assertions for specific elements
    const button = screen.getByRole('button', { name: /switch to dark mode/i });
    expect(button).toBeInTheDocument();
    expect(button.className).toContain('bg-gray-200');
    expect(button.className).toContain('text-gray-800');
    
    const svg = button.querySelector('svg');
    expect(svg).toBeInTheDocument();
    const path = svg.querySelector('path');
    expect(path.getAttribute('d')).toContain('M20.354 15.354A9 9 0 018.646 3.646');
  });
  
  it('calls toggleTheme when clicked', () => {
    // Setup mock return value for useTheme
    const toggleTheme = vi.fn();
    vi.mocked(ThemeContextModule.useTheme).mockReturnValue({ 
      isDark: true, 
      toggleTheme 
    });
    
    render(<ThemeToggle />);
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(toggleTheme).toHaveBeenCalledTimes(1);
  });
});