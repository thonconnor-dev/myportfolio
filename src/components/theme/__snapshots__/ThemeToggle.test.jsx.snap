// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeToggle > renders correctly in dark mode 1`] = `
<div>
  <button
    aria-label="Switch to light mode"
    class="p-2 rounded-full transition-colors duration-300 bg-gray-800 hover:bg-gray-700 text-white"
  >
    <svg
      class="w-6 h-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</div>
`;

exports[`ThemeToggle > renders correctly in light mode 1`] = `
<div>
  <button
    aria-label="Switch to dark mode"
    class="p-2 rounded-full transition-colors duration-300 bg-gray-200 hover:bg-gray-300 text-gray-800"
  >
    <svg
      class="w-6 h-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</div>
`;
